src/general/DayNightCycle.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using Newtonsoft.Json;
+
+[JsonObject(IsReference = true)]
+[UseThriveSerializer]
+public class DayNightCycle
+{
+    [JsonProperty]
+    public DayNightConfiguration LightCycleConfig;
+
+    [JsonProperty]
+    public bool IsEnabled;
+
+    /// <summary>
+    ///   The multiplier used for calculating DayLightPercentage.
+    /// </summary>
+    /// <remarks>
+    ///   This exists as it only needs to be calculated once and
+    ///   the calculation for it is confusing.
+    /// </remarks>
+    [JsonIgnore]
+    private float daytimeMultiplier;
+
+    public DayNightCycle(bool isEnabled)
+    {
+        IsEnabled = isEnabled;
+
+        LightCycleConfig = SimulationParameters.Instance.GetDayNightCycleConfiguration();
+        Time = LightCycleConfig.HoursPerDay / 2;
+
+        float halfPercentage = LightCycleConfig.DaytimePercentage / 2;
+
+        // This converts the percentage in DaytimePercentage to the power of two needed for DayLightPercentage
+        daytimeMultiplier = (float)Math.Pow(2, 1 / halfPercentage);
+
+        if (IsEnabled)
+        {
+            AverageSunlight = EvaluateAverageSunlight(0.5f + halfPercentage)
+                - EvaluateAverageSunlight(0.5f - halfPercentage);
+        }
+        else
+        {
+            AverageSunlight = 1;
+        }
+    }
+
+    [JsonIgnore]
+    public float AverageSunlight { get; private set; }
+
+    /// <summary>
+    ///   The current time in hours
+    /// </summary>
+    [JsonProperty]
+    public float Time { get; private set; }
+
+    [JsonIgnore]
+    public float PercentOfDayElapsed => Time / LightCycleConfig.HoursPerDay;
```

Comment:
If I'm not mistaken, `Time` is only used within this file. As you only use it to define `PercentOfDayElapsed`, you could get rid of it. In particular, you would avoid multiplication/division by `LightCycleConfig.HoursPerDay`.

Reply:
I have it public as it seems like something that would be useful outside the file. For example for an indicator if we ever add one.

Reply:
Sure, but then I would save the value in `PercentOfDayElapsed`, especially since you're basically reconverting Time every time you change it.

Reply:
I also think that we should make the fraction or percent elapsed the main time tracking variable, with the variable controlling how fast that changes based on elapsed time what's tweaked to get the speed right.

Reply:
The main reason time exists is for stuff like clocks, though I guess you are right about it being better to use PercentOfDayElapsed.